improver/psychrometric_calculations/total_precipitable_water.py

```diff
@@ -0,0 +1,147 @@
+# (C) Crown Copyright, Met Office. All rights reserved.
+#
+# This file is part of 'IMPROVER' and is released under the BSD 3-Clause license.
+# See LICENSE in the root of the repository for full licensing details.
+"""Module to generate total precipitable water"""
+
+import iris
+import numpy as np
+from iris.cube import Cube
+from numpy import ndarray
+
+from improver import BasePlugin
+from improver.constants import EARTH_SURFACE_GRAVITY_ACCELERATION, WATER_DENSITY
+
+
+def ensure_no_float_64(cube: Cube) -> Cube:
+    """
+    Convert float64 data and coordinates in the cube to float32 to ensure workflow compatibility.
+    """
+    if cube.data.dtype == np.float64:
+        cube.data = cube.data.astype(np.float32)
+    for coord in cube.coords():
+        if coord.points.dtype == np.float64:
+            coord.points = coord.points.astype(np.float32)
+        if coord.bounds is not None and coord.bounds.dtype == np.float64:
+            coord.bounds = coord.bounds.astype(np.float32)
```

Comment:
I think this plugin should only be checking for data types that it is actually changing. As it doesn't change any of the coordinates, merely copies them, I don't think we should do this here, as it could change a coordinate that needs to be float64.